design/proposals/load_policy/load_policy_validation.md

```diff
@@ -0,0 +1,148 @@
+hosts with empty JWT annotation value, we are failing at JWT authentication request which is bad UX.
+
+## Solution
+For better UX we want to perform policy validation steps in order to identify configuration errors early as possible. 
+
+### Solution - scope
+This solution is creating a generic infrastructure for all policy entities validation.
+And focusing on Host with JWT authenticator annotations as first use case to implement on this new suggested infrastructure.
+
+### User Interface
+TBD
+
+## Design
+The following diagrams, represent a new policy validation layer which will be triggered as part of the load policy request,
+As the final state after the current logic.
+
+
+### Flow Diagrams
+In our use case, Host with JWT authenticator annotations validation:
+1. Conjur admin define JWT authenticator 
+2. Conjur admin load a policy of Host to be authenticated with JWT authenticator
+3. New policy validation layer is triggered and validate if the host is configured correctly 
+4. Host authenticate using JWT authenticator, due to above validation there are no configuration errors at the authenticate request stage
+
+### Sequence Diagrams
+![Load policy validation sequence diagram](load-policy-validation-sequence-diagram.png)
+
+#### Sequence Diagrams - additional info
+1. Policy validation entry point:
+As described in the diagram the validation will be triggered as the final new stage of the current logic (read below),
+   which will call to a new command class `ValidatePolicyConstraints` with tne input of the policy entities 
+   records (see example in the end the of the diagram) from the user body request.
+   
+Load policy current logic - 
+   for the load policy rest requests put/post/patch the current logic in high level is:
+    The current logic: 
+       
+- policy syntax check
+- load records from policy 
+- create temp schema
+- merge new temp schema to main schema 
+- the validation logic will start after the schema merged to main
+
+2. new ValidatePolicyConstraints command class - is responsible to validate all the policy entities 
+   (host,user,webservice) which were sent by the user in load policy request.
+   
+    Note: in this design we will validate only hosts 
+   
+3. new CreatePolicyEntityValidatorsFactory command class - is responsible to create a list of validators as the amount 
+   of the different policy entities types in the records input.
+   
+    Note: in this design we will create only `ValidateHostConstraints` with all the hosts ids in his belly 
+   
+4. new ValidateHostConstraints command class - is responsible to validate a list of host ids which are initialized in his creation.
+   the flow is to fetch the host annotations from the DB (from the main schema after the merge) and to 
+   use the `CreateHostValidatorsFactory` to create hosts authenticators validators.
+   
+5. new CreateHostValidatorsFactory command class - is responsible to create a list of validators as the amount
```

Comment:
same here. you can use `CreateHostValidators`

Reply:
Why? i think it is makes the class a bit more clear. 
by reading the word factory you are already guessing that the class is creating a different kinds of host validators based on some decision logic